### DISCUSSION TITLE
Enable cgroup stats for `misc`

### DISCUSSION THREAD

**Discussion by mythi** [opencontainers/runc#3910] [Ideas]
While the `misc` built-in controller does not have too many resources to manage today, we are working on adding support for [SGX EPC memory](https://lore.kernel.org/linux-sgx/20221202183655.3767674-1-kristen@linux.intel.com/T/#t) to it and for which "per container" stats are very useful.

With OCI / `runc`, I can already set up `misc.max` easily through `config.json`:
```
"unified": {​
    "misc.max": "sgx_epc 42"​
}
```
that gives me (in my setup):
```
cat /sys/fs/cgroup/kubepods.slice/kubepods-besteffort.slice/kubepods-besteffort-pod6df0ca6a_5cf4_4839_82dc_7337d00b71ed.slice/cri-containerd-d7cbd4f59bb80c46b39656bcfad0e3a4c16ec62a3919d5d559f2b29255191437.scope/
misc.max
sgx_epc 42
```

`misc.current` returns the per container usage of `sgx_epc`.

I'm looking for feedback on whether it would make sense to have `misc` stats enabled through libcontainer `CgroupManager` which would then allow me to update cAdvisor or in other places to get `misc` stats.

Thoughts?

**Comment by eero-t** [comment]
Somewhat related: https://github.com/opencontainers/runc/issues/3849

**Comment by mythi** [comment]
The code was added to libcontainer.
